internal/converter/converter.go

```diff
@@ -0,0 +1,56 @@
+// Package converter implements functionality for converting and compressing images.
+package converter
+
+import (
+	"errors"
+	"image"
+	"image/jpeg"
+	"image/png"
+	"mime/multipart"
+	"os"
+	"time"
+)
+
+const tempPath = "temp/"
+
+// Converter converts and compresses images.
+type Converter struct{}
+
+// NewConverter creates new converter.
+func NewConverter() *Converter {
+	return &Converter{}
+}
+
+//Convert converts and compresses the given image file according to the target format and compression ratio.
+func (c *Converter) Convert(file multipart.File, targetFormat string, ratio int) (*os.File, error) {
```

Comment:
Converter is not responsible for storing images, only for converting them. Storing converted image is a part of business logic

Reply:
> Converter is not responsible for storing images, only for converting them. Storing converted image is a part of business logic
Converter creates an intermediate file, which is then deleted in the business logic by its filename